core/domain/stats_services.py

```diff
@@ -212,6 +213,165 @@ def handle_stats_creation_for_new_exp_version(
+            # Modify occurences in exploration issues.
+            if stats_models.ISSUE_TYPE_KEYNAME_MAPPING[
+                    exp_issue.issue_type] == 'state_names':
+                state_names = exp_issue.issue_customization_args['state_names'][
+                    'value']
+                exp_issue.issue_customization_args['state_names']['value'] = [
+                    new_state_name
+                    if state_name == old_state_name else state_name
+                    for state_name in state_names]
+            else:
+                exp_issue.issue_customization_args['state_name']['value'] = (
+                    new_state_name)
+            playthrough_ids.extend(exp_issue.playthrough_ids)
+
+        # Modify occurences in playthroughs.
+        playthroughs = get_playthroughs_multi(playthrough_ids)
+        for idx, playthrough in enumerate(playthroughs):
+            if stats_models.ISSUE_TYPE_KEYNAME_MAPPING[
+                    playthrough.issue_type] == 'state_names':
+                state_names = playthrough.issue_customization_args[
+                    'state_names']['value']
+                playthrough.issue_customization_args['state_names']['value'] = [
+                    new_state_name
+                    if state_name == old_state_name else state_name
+                    for state_name in state_names]
+            else:
+                playthrough.issue_customization_args['state_name']['value'] = (
+                    new_state_name)
+            for idx1, action in enumerate(playthrough.playthrough_actions):
+                if action.action_customization_args['state_name']['value'] == (
+                        old_state_name):
+                    playthroughs[idx].playthrough_actions[
+                        idx1].action_customization_args['state_name'][
+                            'value'] = new_state_name
+
+        update_playthroughs_multi(playthrough_ids, playthroughs)
+
+    # Handling reverts.
+    if change_list[0]['cmd'] == 'AUTO_revert_version_number':
+        revert_to_version = change_list[0]['version_number']
+        # TODO(pranavsid98): Convert below two lines to use get_by_id() instead.
+        old_exp_issues_instance = stats_models.ExplorationIssuesModel.get_model(
+            exploration.id, revert_to_version)
+        old_exp_issues = get_exp_issues_from_model(old_exp_issues_instance)
+        exp_issues.unresolved_issues = old_exp_issues.unresolved_issues
+
+    exp_issues.exp_version += 1
+
+    create_exp_issues_model(exp_issues)
+
+
+def get_playthroughs_multi(playthrough_ids):
+    """Retrieves multiple Playthrough domain objects.
+
+    Args:
+        playthrough_ids: list(str). List of playthrough IDs.
+
+    Returns:
+        list(Playthrough). List of playthrough domain objects.
+    """
+    playthrough_instances = stats_models.PlaythroughModel.get_multi(
+        playthrough_ids)
+    # TODO(pranavsid98): Replace below lines with get_from_model.
```

Comment:
Why not go ahead and resolve this todo?

Reply:
After 1.2.3 PR

Reply:
Ah ok, thanks. In the future, for stuff like this, please include the PR number in the TODO comment or in the GitHub comment thread.